.github/workflows/publish.yml

```diff
@@ -8,7 +8,100 @@ on:
 jobs:
   tests:
     uses: ./.github/workflows/test.yml
-  publish-release:
+  
+  build-installer:
+    strategy:
+      fail-fast: false
+      matrix:
+        job:
+          - { target: x86_64-unknown-linux-gnu, exe: amd64-linux, os: ubuntu-latest}
+          - { target: x86_64-apple-darwin, exe: macos, os: macos-latest}
+          - { target: x86_64-pc-windows-gnu, exe: windows.exe, os: windows-latest}    
+    runs-on: ${{ matrix.job.os }}
+    defaults: 
+      run:
+        working-directory: ./micro-rdk-installer
+        shell: bash
+    steps:
+    - name: Checkout main branch code
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 2
+    - name: Set toolchain
+      uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        toolchain: 1.70.0
+        override: true
+        target: ${{ matrix.job.target }}
+        default: true
+    - name: Install x86_64 linux dependencies
+      if: ${{ matrix.job.os == 'ubuntu-latest' }}
+      run: |
+        sudo apt-get install libudev-dev pkg-config libssl-dev
+    - name: Install MacOS dependencies
+      if: ${{ matrix.job.os == 'macos-latest' }}
+      run: |
+        brew install coreutils
+        export PATH="/usr/local/opt/coreutils/libexec/gnubin:$PATH"
+    - name: Build installer binary
+      run: |
+        cargo build --release --features static_tls --target ${{ matrix.job.target }} --locked
+    - name: Rename result
+      run: |
+        rm target/${{ matrix.job.target }}/release/micro-rdk-installer.d
+        if [[ $(find . -wholename 'target/${{ matrix.job.target }}/release/micro-rdk-installer*' -printf c | wc -c) == "1" ]]; then
+            cp target/${{ matrix.job.target }}/release/micro-rdk-installer* micro-rdk-installer-${{ matrix.job.exe }}
+        else
+            exit 1
+        fi
+    - name: Check release type
+      id: check-tag
+      run: |
+        if echo ${{ github.event.ref }} | grep -Eq '^refs/tags/v.*rc[0-9]{1}$'; then
+              echo "match=true" >> $GITHUB_OUTPUT
+        else
+              echo "match=false" >> $GITHUB_OUTPUT
+        fi
+    - name: Upload release
+      uses: actions/upload-artifact@v2
+      with:
+        name: micro-rdk-installer-${{ matrix.job.exe }}
+        path: micro-rdk-installer-${{ matrix.job.exe }}
+  
+  build-installer-arm64-linux:
```

Comment:
ideally add comment here saying why this needs its own job (vs the other 3 architectures which share a job)